src/wp-admin/upload.php

```diff
@@ -128,13 +128,14 @@
 	$_SERVER['REQUEST_URI'] = remove_query_arg( array( 'message' ), $_SERVER['REQUEST_URI'] );
 }
 
-$mode  = get_user_option( 'media_library_mode', get_current_user_id() ) ? get_user_option( 'media_library_mode', get_current_user_id() ) : 'grid';
 $modes = array( 'grid', 'list' );
 
 if ( isset( $_GET['mode'] ) && in_array( $_GET['mode'], $modes, true ) ) {
 	$mode = $_GET['mode'];
 	update_user_option( get_current_user_id(), 'media_library_mode', $mode );
-}
+} else {
+	$mode = get_user_option( 'media_library_mode' ) ?: 'grid';
```

Comment:
Since short ternaries are not allowed in the WP coding standards.

```suggestion
	$mode = get_user_option( 'media_library_mode' );
	$mode = $mode ? $mode : 'grid';
```